includes/class-wc-payments-webhook-processing-service.php

```diff
@@ -740,11 +741,11 @@ private function get_order_from_event_body( $event_body ) {
 				$order = $this->wcpay_db->order_from_order_id( $order_id );
 			} elseif ( ! empty( $event_object['invoice'] ) ) {
 				// If the payment intent contains an invoice it is a WCPay Subscription-related intent and will be handled by the `invoice.paid` event.
-				return false;
+				return null;
 			}
 		}
 
-		if ( ! $order ) {
+		if ( ! $order instanceof \WC_Order ) {
```

Comment:
In 99% of cases, this works fine. However, if you take a closer look at this line: `this->wcpay_db->order_from_order_id`, you'll see that `order_from_order_id` can return a `WC_Order_Refund` (according to the docblock). Since `\WC_Order` is not an instance of `WC_Order_Refund`, this would make the following statement false and potentially cause issues. Instead, you could use the abstract class `WC_Abstract_Order` for the comparison. Both `WC_Order` and `WC_Order_Refund` extend this class, which would make the statement true.

Reply:
Thanks for your suggestion to use the `WC_Abstract_Order` class @zmaglica. For context,  I was initially using [it]([b93afec](https://github.com/Automattic/woocommerce-payments/pull/9761/commits/b93afec06e5e94c8a8d53da101f6c0afdba02b30)), but decided to go with the current approach after thinking that this flow is based on the `succeeded` and `failed` events of the `payment_intent`. In theory, we should never find an instance of `WC_Order_Refund`, but better to be safe than sorry. I'm going with the stricter approach 👍

Reply:
After thinking about it more, I believe it’s better not to use `WC_Abstract_Order` here. I get that we could potentially be dealing with a `WC_Order_Refund`, since that’s a possible return type of `this->wcpay_db->order_from_order_id`. But I don’t think that’s something we should expect, and I feel like throwing an exception would make more sense in this case.

Here’s my reasoning: we’re handling a payment_intent event from Stripe. If we don’t find an order linked to the `payment_intent` ID but end up with a `WC_Order_Refund` instead, I think it would be wrong to move forward with processing the refund object. Raising an exception feels like the right approach here.

Also, refunds in Stripe are their own separate entities and are dealt with in a different flow, so this situation doesn’t align with what we’d expect. That’s my take on it, but I’m happy to hear other thoughts or be corrected if I’m missing something.

Reply:
Got it. Let's stick with your original plan. If another type of WC_Order is encountered, an exception will be thrown, which is ideal since no other order types should exist.